rmf_site_editor/src/main_menu.rs

```diff
@@ -66,16 +66,20 @@ fn egui_ui(
             ui.add_space(10.);
 
             ui.horizontal(|ui| {
-                if ui.button("View demo map").clicked() {
-                    _load_workspace.send(LoadWorkspace::Data(WorkspaceData::LegacyBuilding(
-                        demo_office(),
-                    )));
+                if ui.button("New site").clicked() {
+                    _load_workspace.send(LoadWorkspace::Data(WorkspaceData::Site(empty_site())));
```

Comment:
I gave it a quick spin and it seems slightly hairy because the `CreateNewWorkspace` event requires the `AppState` to be set otherwise it will return an error. Naively setting the AppState before sending the event also seems to not work, probably because the state update was not flushed before the event reader ran. Therefore unless we reorder systems to make sure the app state was correctly set and flushed before the event reader runs the `CreateNewWorkspace` approach will not work.

Still embedding another demo map is not ideal. I suspect if we set the state and made sure the `CreateNewWorkspace` events are processed in a separate stage it should work. Another idea could be to just use the site constructor that is used inside the `CreateNewWorkspace` event handler for this, and just dispatch a `LoadSite` event.
It's a bit hacky but we need to refactor states soon anyway when upgrading the bevy version so :shrug:

@arjo129 is right and we should definitely merge that PR first.
I'll also open up a quick PR to fix CI now that `let else` support has been added to `rustfmt` to make it green again.